### ISSUE TITLE
Enable simulators in addition to H6303, get M6800 working

### ISSUE THREAD

**Issue by nickd4** [dg1yfe/sim68xx#5]
For my experiments I wanted 6800 simulation, so I tried to compile the M6800 arch but it seemed to have a lot of bitrot. As a result I started by enabling the other simulators (all but M6800) and fixing some very small bitrot so they could compile. And then I copied the H6301 over the M6800 and deleted its internal registers. I also created a basic ASCI, given that the M6800 does not have an internal one.

At this point I could run an Altair 680 ROM and load Altair 680 BASIC. However, things are a bit of a mess as I can clearly see that copying the H6301 opcode decoder over the M6800 one was too brutal since the changeset shows only a few changes necessary for fixing bitrot, and the rest of the changes are adding new instructions that shouldn't be there. Hence I'm creating an issue rather than a PR.

Also, my ASCI simulation is kludgey. As things are now, I could re-enable M6800 SCI peripheral and put it at 0xf000 where the ASCI is located on Altair 680. I didn't do it that way originally because I was adding the ASCI to H6301 and so I had to duplicate the internal peripherals decoding logic in a kludgey way to put my ASCI at 0xf000. After doing that I realized that Altair 680 BASIC relies on RAM in the internal peripherals address space and so it can't run on H6301 and required me to fix M6800 first.

In the longer term I think the best way would be to allow the board logic to intercept RAM accesses so it can have its own peripherals outside of the internal peripheral register space, but another way would be to implement an "external" space using a copy of the same code that does the "internal" space. Then external peripherals would need to be located contiguously in memory (that's usually the case anyway).

To see my version, with a file `/src/boards/basic.s19` containing the Altair 680 BASIC and instructions for launch at the top of this `s19` file, please have a look here: https://github.com/nickd4/sim68xx.git

**Comment by dg1yfe** [comment]
Hi nickd4,
nice to see that you have taken interest in this old code I found on the web more than 10 years ago. I myself have only been interested in getting the 6301/6303 stuff to run (and found and fixed a lot of issues along the way). I never touched or tested the other simulators.
I currently have zero time to do anything with this project. Nevertheless I would merge fixes and enhancements if I'm given a Pull Request. For more I simply do not have the time to spare at the moment - sorry.

**Comment by nick-lifx** [comment]
It's good to hear you would merge PRs. Yes I certainly understand that this repo isn't a priority, on the other hand it's good that you are at least checking for activity on your github repos :) Some people entered PRs against quick one off projects I did to solve a problem ages ago, and it honestly never occurred to me to check for such things :)

I am a bit closer to a PR as I have now backed out added 6301 instructions in the 6800 arch and then referring to an opcode map I found somewhere online I deleted most 16 bit instructions and some Hitachi specific ones. So I think the opcode map is correct. But it has some fields I did not really look at and I think these are the bytes and cycles of the illegal opcodes, eg some have an addressing mode so they create multi cycle and multi byte no ops I think. But lacking hard information I did not fill these in.

However the 6800 arch won't do anything with no I/O so I would need to tidy this up before any PR.
